core/api/usermanager/v1/usermanager_proto/usermanager.proto

```diff
@@ -0,0 +1,156 @@
+  // user contains data which will be applied to the user.
+  keytransparency.v1.proto.User user = 2;
+  // update_mask specifies which fields of user to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 3;
+}
+
+// BatchCreateUserRequest creates multiple users all at once.
+message BatchCreateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // users is the set of users to create.
+  repeated keytransparency.v1.proto.User users = 3;
+  // add_signing_keys specifies whether to add this service's signing_keys to the set of authorized_keys.
+  // This must be set to true if any further operations from this API are meant to succeed.
+  // If set to false, there must be at least one key in authorized_keys.
+  bool add_signing_keys = 4;
+}
+
+// BatchCreateUserRequest creates multiple users at once.
+message BatchCreateUserResponse {
+  // users returns the list of created users.
+  repeated keytransparency.v1.proto.User users = 2;
+}
+
+// BatchUpdateUserRequest creates multiple users all at once.
+message BatchUpdateUserRequest {
+   // domain_id identifies the domain.
+   string domain_id = 1;
+   // app_id identifies the application.
+   string app_id = 2;
+  // users is the set of users to create.
+  repeated keytransparency.v1.proto.User users = 3;
+  // update_mask specifies which fields of user to update.
+  // For example: "data" or "authorized_keys"
+  google.protobuf.FieldMask update_mask = 4;
+}
+
+// BatchUpdateUserRequest creates multiple users at once.
+message BatchUpdateUserResponse {
+  // users returns the list of created users.
+  repeated keytransparency.v1.proto.User users = 2;
+}
+
+
+// The UserManagerService holds a set of signing keys which can be used to
+// sign user updates.
+// - This service must be registered with the server in order to submit user
+// create or update requests.
+// - The update requests this service generates will only be valid for users
+// that have added this service's signing keys to their list of authorized_keys.
+// - This service should carefully controll and log access to it's APIs.
+// - This service should rotate its signing keys periodically.
+//   New signing keys should be published far enough in advance that they can be added
+//   by everyone before they are used to avoid service disruption.
+//
+// Resources:
+// - domains/apps/keyset
+//   Each combination of domain_id and app_id has it's own keyset.
+// - domains/apps/users
+//   Users can be created and updated through this endpoint.
+service UserManagerService {
+
+  // GetKeySet returns a list of public keys (a keyset) that corresponds to the signing keys
+  // this service has for a given domain and app.
+  rpc GetKeySet (GetKeySetRequest) returns (keymaster.proto.KeySet) {
+    option (google.api.http) = {
+      get: "/usermanager/v1/domains/{domain_id}/apps/{app_id}/keyset"
```

Comment:
Since you have an implied 1:1 relationship between KeySets and Apps would it make sense to return them as part of the App entity? That would mean you have a "GetApp" method instead of "GetKeySet".

"No" is perfectly fine answer here, specially if there are motives for the design that I might be missing.

Reply:
Interesting idea.  
If the App type defined in usermanager is allowed to be different than the main API, then yes, we could do that.  It is possible, however, that we may want to make the keyset end point more broadly available than the other info we might want to put in GetApp